### ISSUE TITLE
Windows: Compilation process locks files, cannot rerun SAM Local

### ISSUE THREAD

**Issue by roman-svystun** [aws/aws-toolkit-jetbrains#694]
**Describe the bug**
After making any change to a class file, running Lambda Local Configuration results in `Access is denied` errors

**To reproduce**
1. Create and execute a Lambda Local configuration
2. Make any change to any class files
3. Attempt to rerun the Lambda Local configuration

**Expected behavior**
Intellij Idea compilation process should release any file locks. User should be able to edit class files and rerun Lambda configurations.

**Your Environment**
 - OS: Windows 10 Enterprise
 - JetBrains' Product: Idea Ultimate 2018.3
 - JetBrains' Product Version: 183.4886.3
 - Toolkit Version: 1.0

**Additional context**
Stack trace for entire work-flow:
```
2018-12-11 13:22:41,572 [  70800]   INFO - ij.compiler.impl.CompileDriver - COMPILATION STARTED (BUILD PROCESS) 
2018-12-11 13:22:42,085 [  71313]   INFO - j.compiler.server.BuildManager - BUILDER_PROCESS [stdout]: Build process started. Classpath: C:/progs/JetBrains/IntelliJ IDEA 183.2153.8/lib/jps-launcher.jar;C:/Java/jdk1.8.0_151/lib/tools.jar;C:/progs/JetBrains/IntelliJ IDEA 183.2153.8/lib/optimizedFileManager.jar 
2018-12-11 13:22:42,115 [  71343]   INFO - j.compiler.server.BuildManager - BUILDER_PROCESS [stderr]: SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder". 
2018-12-11 13:22:42,115 [  71343]   INFO - j.compiler.server.BuildManager - BUILDER_PROCESS [stderr]: SLF4J: Defaulting to no-operation (NOP) logger implementation 
2018-12-11 13:22:42,115 [  71343]   INFO - j.compiler.server.BuildManager - BUILDER_PROCESS [stderr]: SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details. 
2018-12-11 13:22:45,539 [  74767]   INFO - lij.compiler.impl.CompilerUtil - 	COMPILATION FINISHED (BUILD PROCESS); Errors: 0; warnings: 0 took 3974 ms: 0 min 3sec 
2018-12-11 13:22:45,570 [  74798]   INFO - s.CompilerReferenceServiceBase - backward reference index reader is opened 
2018-12-11 13:22:45,866 [  75094]   INFO - j.compiler.server.BuildManager - BUILDER_PROCESS [stdout]: Build process started. Classpath: C:/progs/JetBrains/IntelliJ IDEA 183.2153.8/lib/jps-launcher.jar;C:/Java/jdk1.8.0_151/lib/tools.jar;C:/progs/JetBrains/IntelliJ IDEA 183.2153.8/lib/optimizedFileManager.jar 
2018-12-11 13:22:45,895 [  75123]   INFO - j.compiler.server.BuildManager - BUILDER_PROCESS [stderr]: SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder". 
2018-12-11 13:22:45,895 [  75123]   INFO - j.compiler.server.BuildManager - BUILDER_PROCESS [stderr]: SLF4J: Defaulting to no-operation (NOP) logger implementation 
2018-12-11 13:22:45,895 [  75123]   INFO - j.compiler.server.BuildManager - BUILDER_PROCESS [stderr]: SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details. 
2018-12-11 13:23:05,347 [  94575]   WARN - ntellij.ui.popup.AbstractPopup - huge popup requested: 2904 x 30 
2018-12-11 13:23:06,661 [  95889]   WARN - ntellij.ui.popup.AbstractPopup - huge popup requested: 2904 x 30 
2018-12-11 13:23:32,965 [ 122193]   INFO - ij.compiler.impl.CompileDriver - COMPILATION STARTED (BUILD PROCESS) 
2018-12-11 13:23:32,968 [ 122196]   INFO - j.compiler.server.BuildManager - Using preloaded build process to compile E:\ws\lambda_api_proxy 
2018-12-11 13:23:35,592 [ 124820]   INFO - lij.compiler.impl.CompilerUtil - 	COMPILATION FINISHED (BUILD PROCESS); Errors: 3; warnings: 0 took 2634 ms: 0 min 2sec 
2018-12-11 13:23:35,593 [ 124821]   WARN - .execution.sam.SamInvokeRunner - Failed to create Lambda package 
java.util.concurrent.CompletionException: java.lang.RuntimeException: Compilation completed with errors.
java: E:\ws\lambda_api_proxy\target\classes\com\example\aws\vpc\proxy\examplews\AbstractProxyAPICallsToExample$TrustAllHostNameVerifier.class (Access is denied)
java: E:\ws\lambda_api_proxy\target\classes\com\example\aws\vpc\proxy\examplews\AbstractProxyAPICallsToExample$1.class (Access is denied)
java: E:\ws\lambda_api_proxy\target\classes\com\example\aws\vpc\proxy\examplews\AbstractProxyAPICallsToExample.class (Access is denied)
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:647)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:632)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977)
	at software.aws.toolkits.jetbrains.services.lambda.java.JavaLambdaPackager$createPackage$1.finished(JavaLambdaPackager.kt:62)
	at com.intellij.compiler.CompilerManagerImpl$ListenerNotificator.finished(CompilerManagerImpl.java:529)
	at com.intellij.compiler.impl.CompileDriver.a(CompileDriver.java:500)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:719)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:668)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:363)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.lang.RuntimeException: Compilation completed with errors.
java: E:\ws\lambda_api_proxy\target\classes\com\example\aws\vpc\proxy\examplews\AbstractProxyAPICallsToExample$TrustAllHostNameVerifier.class (Access is denied)
java: E:\ws\lambda_api_proxy\target\classes\com\example\aws\vpc\proxy\examplews\AbstractProxyAPICallsToExample$1.class (Access is denied)
java: E:\ws\lambda_api_proxy\target\classes\com\example\aws\vpc\proxy\examplews\AbstractProxyAPICallsToExample.class (Access is denied)
	at software.aws.toolkits.jetbrains.services.lambda.java.JavaLambdaPackager$createPackage$1.finished(JavaLambdaPackager.kt:63)
	... 19 more
2018-12-11 13:23:35,601 [ 124829]   INFO - s.CompilerReferenceServiceBase - backward reference index reader is opened 
2018-12-11 13:23:35,882 [ 125110]   INFO - j.compiler.server.BuildManager - BUILDER_PROCESS [stdout]: Build process started. Classpath: C:/progs/JetBrains/IntelliJ IDEA 183.2153.8/lib/jps-launcher.jar;C:/Java/jdk1.8.0_151/lib/tools.jar;C:/progs/JetBrains/IntelliJ IDEA 183.2153.8/lib/optimizedFileManager.jar 
2018-12-11 13:23:35,911 [ 125139]   INFO - j.compiler.server.BuildManager - BUILDER_PROCESS [stderr]: SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder". 
2018-12-11 13:23:35,911 [ 125139]   INFO - j.compiler.server.BuildManager - BUILDER_PROCESS [stderr]: SLF4J: Defaulting to no-operation (NOP) logger implementation 
2018-12-11 13:23:35,911 [ 125139]   INFO - j.compiler.server.BuildManager - BUILDER_PROCESS [stderr]: SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details. 
```

**Comment by awschristou** [comment]
Thank you for reporting the issue. A fix has been submitted, which should address this issue in a future release of the toolkit.

**Comment by lrevnic** [comment]
@awschristou , do you know when the fix will be provided? I'm writing an article on the toolkit and I would like to mention the plans of fixing this issue. Thanks!

**Comment by rli** [comment]
@lrevnic This was fixed in PR #697 and has been merged to master.
